ShortestPath/ShortestPath/DijkstraAlgorithm.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace ShortestPath
+{
+    public class DijkstraAlgorithm
+    {
+        public Dictionary<string, Node> ExtractedNodes = new Dictionary<string, Node>();
+        public IHeap heap;
+
+        public DijkstraAlgorithm(HeapFactory factory)
+        {
+            heap = factory.GetHeap();
+        }
+
+        public void FindShortestPath(Network network, string origin, string destination)
+        {
+            try
+            {
+                InitializeHeap(network);
+                UpdateNodesProperties(network, destination);
+                List<string> ShortestPath = new List<string>();
+                ShortestPath.Add(origin);
+                string InitialNode = origin;
+
+                while (true)
+                {
+                    var FollowingNode = ExtractedNodes[InitialNode].Successor;
+                    InitialNode = FollowingNode;
+                    ShortestPath.Add(FollowingNode);
+                    if (InitialNode == destination) { break; }
+                }
+                Print(ShortestPath);
+            }
+            catch
+            {
+                Console.WriteLine("There is no path between the given Origin - Destination");
+            }
+        }
+
+        private void InitializeHeap(Network network)
+        {
+            heap = new Heap(network.NetworkNodes);
```

Comment:
Are you sure that you need to add all the nodes with infinite cost at the beginning?